internal/datacoord/move_binlog_path_helper.go

```diff
@@ -0,0 +1,64 @@
+package datacoord
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/milvus-io/milvus/internal/kv"
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/proto/datapb"
+	"go.uber.org/zap"
+)
+
+type MoveBinlogPathHelper struct {
+	kv   kv.TxnKV
+	meta *meta
+}
+
+func NewMoveBinlogPathHelper(kv kv.TxnKV, meta *meta) *MoveBinlogPathHelper {
+	return &MoveBinlogPathHelper{
+		kv:   kv,
+		meta: meta,
+	}
+}
+
+func (h *MoveBinlogPathHelper) Execute() error {
+	segmentIds := h.meta.GetSegments()
+
+	if len(segmentIds) == 0 {
+		log.Debug("there's no segments in meta")
+	} else if len(segmentIds) == 1 {
+		log.Debug("there's 1 segment's binlogs to move", zap.Int64("segmentID", segmentIds[0]))
+	} else {
+		log.Debug(fmt.Sprintf("there are %d segments' binlogs to move", len(segmentIds)))
+	}
+
+	for _, id := range segmentIds {
+		m := make(map[UniqueID][]string)
+		p := path.Join(Params.SegmentBinlogSubPath, strconv.FormatInt(id, 10)) + "/" // prefix/id/ instead of prefix/id
+		_, values, err := h.kv.LoadWithPrefix(p)
+		if err != nil {
+			log.Error("failed to load prefix", zap.String("prefix", p), zap.Error(err))
+			return err
+		}
+		for _, v := range values {
+			tMeta := &datapb.SegmentFieldBinlogMeta{}
+			if err := proto.UnmarshalText(v, tMeta); err != nil {
+				log.Error("failed to unmarshal", zap.Error(err))
```

Comment:
this may happen if we Execute to half way and the datacoord crashed. some of the logs use old path, the others use new format?

Reply:
Yes. When the datacoord restarts, we will continue to transfer these paths to new format.